l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,483 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {IConsensusRegistry} from "./interfaces/IConsensusRegistry.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @title ConsensusRegistry
+/// @dev Manages consensus nodes and committees for the L2 consensus protocol,
+/// owned by Matter Labs Multisig. Nodes act as both validators and attesters,
+/// each playing a distinct role in the consensus process. This contract facilitates
+/// the rotation of validator and attester committees, which represent a subset of nodes
+/// expected to actively participate in the consensus process during a specific time window.
+contract ConsensusRegistry is IConsensusRegistry, Ownable2Step {
+    /// @dev An array to keep track of node owners.
+    address[] public nodeOwners;
+    /// @dev A mapping of node owners => nodes.
+    mapping(address => Node) public nodes;
+    /// @dev A mapping for enabling efficient lookups when checking whether a given attester public key exists.
+    mapping(bytes32 => bool) public attesterPubKeyHashes;
+    /// @dev A mapping for enabling efficient lookups when checking whether a given validator public key exists.
+    mapping(bytes32 => bool) public validatorPubKeyHashes;
+    /// @dev Counter that increments with each new commit to the attester committee.
+    uint32 private attestersCommit;
+    /// @dev Counter that increments with each new commit to the validator committee.
+    uint32 private validatorsCommit;
```

Comment:
Better to make it public, in case we would want to implement`getAttesterCommittee`/`getValidatorCommittee` offchain 
```suggestion
    /// @dev Counter that increments with each new commit to the attester committee.
    uint32 public attestersCommit;
    /// @dev Counter that increments with each new commit to the validator committee.
    uint32 public validatorsCommit;
```